spark3/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java

```diff
@@ -75,25 +71,15 @@ protected BaseProcedure(TableCatalog tableCatalog) {
   }
 
   // we have to parse both namespace and name as they may be quoted
-  protected Identifier toIdentifier(String namespaceAsString, String name) {
-    String[] namespaceParts = parseMultipartIdentifier(namespaceAsString);
-
-    String[] nameParts = parseMultipartIdentifier(name);
-    Preconditions.checkArgument(nameParts.length == 1, "Name must consist of one part: %s", name);
-
-    return Identifier.of(namespaceParts, nameParts[0]);
-  }
-
-  private String[] parseMultipartIdentifier(String identifierAsString) {
+  protected Identifier toIdentifier(String identifier) {
+    Spark3Util.CatalogAndIdentifier catalogAndIdentifier;
     try {
-      ParserInterface parser = spark.sessionState().sqlParser();
-      Seq<String> namePartsSeq = parser.parseMultipartIdentifier(identifierAsString);
-      String[] nameParts = new String[namePartsSeq.size()];
-      namePartsSeq.copyToArray(nameParts);
-      return nameParts;
+      catalogAndIdentifier = Spark3Util.catalogAndIdentifier(spark, identifier, tableCatalog);
     } catch (ParseException e) {
-      throw new RuntimeException("Couldn't parse identifier: " + identifierAsString, e);
+      throw new IllegalArgumentException("Cannot parse identifier", e);
     }
+
+    return catalogAndIdentifier.identifier();
```

Comment:
We should validate the resolved catalog is `tableCatalog` to make sure we don't modify tables in other catalogs.

Reply:
I didn't want that to be a general requirement, But I could add it here. Ie if we want other procedures which operate on tables in other catalogs

Reply:
Well, maybe the validation should not happen here. We could call it `toCatalogAndIdentifier` and validate above.

Reply:
That's a good idea

Reply:
Looks like this validation is still here?

Reply:
Ah yes I got very confused. I think we should just let this be for now and remove the check. I think instead "loadSparkTable" can have the check since it assumes that we are loading out of this catalog.

Reply:
Sorry I flipped back again, i'm just going to not put the condition here. We'll get an error in "loadSparkTable" which won't have the type argument but should be clear enough?

Reply:
The check definitely needs to happen somewhere, and the error should be specific: `"Cannot run stored procedure %s.%s: %s is not in catalog %s", catalogName, procName, tableName, catalogName`.

I don't see a way for `loadSparkTable` to catch that the original identifier was actually for a different catalog because the catalog for this identifier is discarded.